subsys/net/lib/nrf_cloud/coap/src/nrf_cloud_coap_transport.c

```diff
@@ -181,28 +182,43 @@ static int update_configured_info_sections(const char * const app_ver)
 		return -EALREADY;
 	}
 
-	struct nrf_cloud_svc_info_fota fota = {0};
-	struct nrf_cloud_svc_info_ui ui = {0};
-	struct nrf_cloud_svc_info svc_inf = {
-		.ui = &ui,
-		.fota = &fota
-	};
-	struct nrf_cloud_modem_info mdm_inf = {
-		.application_version = app_ver
-	};
-	struct nrf_cloud_device_status dev_status = {
-		.modem = &mdm_inf,
-		.svc = &svc_inf,
-	};
-	int err = nrf_cloud_shadow_info_enabled_sections_get(&dev_status);
+	/* Create a JSON object to contain shadow info sections */
+	NRF_CLOUD_OBJ_JSON_DEFINE(info_obj);
+	int err = nrf_cloud_obj_init(&info_obj);
 
-	if (err == -ENODEV) {
-		/* Nothing to send */
-		err = 0;
-	} else {
-		err = nrf_cloud_coap_shadow_device_status_update(&dev_status);
+	if (err) {
+		LOG_ERR("Failed to initialize object: %d", err);
+		return err;
+	}
+
+	/* Encode the enabled info sections */
+	err = nrf_cloud_enabled_info_sections_json_encode(info_obj.json, app_ver);
```

Comment:
Would it make more sense for us to just pass `info_obj`, and name the function `nrf_cloud_enabled_info_sections_encode` and return -ENOTSUP for anything other than JSON? I know we've used that pattern in the past

Reply:
if this wasn't in the internal codec, yes that would be preferred.
this function is also used by the MQTT code, so it was simpler to keep it as a cJSON *.